### PR TITLE
fix split to support zero split length

### PR DESCRIPTION
**Pull request by neginraoof**
ONNX split supports split lengths of zero for empty tensors.
This PR adds support for this case.

### CODE REVIEW

**Comment by neginraoof** [comment]
cc @hariharans29 for review. Thanks.

File: onnxruntime/test/providers/cpu/tensor/split_op_test.cc [hunk 1]
```diff
@@ -431,7 +431,7 @@ TEST(SplitOperatorTest, InvalidValueInSplitAttribute) {
                               5.f, 6.f,
                               7.f, 8.f}};
 
-  std::vector<int64_t> splits{1, 0, 3};  // 0 is not valid
+  std::vector<int64_t> splits{1, -1, 3};  // 0 is not valid
   outputs.push_back({{1, 2}, {1.f, 2.f}});
```
Comment (skottmckay) [inline onnxruntime/test/providers/cpu/tensor/split_op_test.cc]: Comment needs updating.

Please add a unit test for your change as well to validate that a length of 0 is handled correctly.

**Comment by hariharans29** [comment]
LGTM. Could you please add a unit test as Scott requested ? Thanks.

File: onnxruntime/test/providers/cpu/tensor/split_op_test.cc (hunk 1 above)
Comment (neginraoof) [reply onnxruntime/test/providers/cpu/tensor/split_op_test.cc]: Thanks!

File: onnxruntime/test/providers/cpu/tensor/split_op_test.cc [hunk 2]
```diff
@@ -431,13 +431,27 @@ TEST(SplitOperatorTest, InvalidValueInSplitAttribute) {
...
 
   RunTest<float>(axis, splits, input, outputs, false, true, "Invalid value in 'split' attribute");  //TensorRT parser: Assertion failed: axis != BATCH_DIM
 }
 
+TEST(SplitOperatorTest, ZeroValueInSplitAttribute) {
+  const int64_t axis = 0;
+  std::vector<ShapeAndFloatData> outputs;
+  // input shape and data
+  ShapeAndFloatData input = CreateInput({0});
+
```
Comment (skottmckay) [inline onnxruntime/test/providers/cpu/tensor/split_op_test.cc]: nit: would maybe be a little easier to understand the expected behavior if there was some actual data in the input. e.g. if the input had 2 values, and splits was { 1, 0, 1 } you could see what the zero equates to in the output. Not quite as clear when splitting nothing into 3 lots of nothing.

**Comment by skottmckay** [comment]
This is failing for the CIs that use CUDA with the following error

> 1: [ RUN      ] SplitOperatorTest.ZeroValueInSplitAttribute
> 1: 2020-01-31 01:02:09.213622034 [E:onnxruntime:Default, sequential_executor.cc:183 Execute] Non-zero status code returned while running Split node. Name:'node1' Status Message: alloc failed
> 1: 2020-01-31 01:02:09.213654034 [E:onnxruntime:Default, provider_test_utils.cc:457 ExecuteModel] Run failed with status: Non-zero status code returned while running Split node. Name:'node1' Status Message: alloc failed
> 1: /onnxruntime_src/onnxruntime/test/providers/provider_test_utils.cc:458: Failure
> 1: Value of: status.IsOK()
> 1:   Actual: false
> 1: Expected: true
> 1: Non-zero status code returned while running Split node. Name:'node1' Status Message: alloc failed
> 1: [  FAILED  ] SplitOperatorTest.ZeroValueInSplitAttribute 

I don't think the CUDA logic for Split handles all the splits being empty. I also don't think it needs to as that doesn't seem like a realistic scenario, so instead of correcting that I'd suggest trying changing the unit test to have some actual data in the input instead.

**Pull request closed without merging**